### DISCUSSION TITLE
.Spec file can build on my computer but cannot build on another computer

### DISCUSSION THREAD

**Discussion by isisgonzalez183** [pyinstaller/pyinstaller#6784] [Help] [answered]
Hello all,
So I created a spec file to build an executable. The executable I built can run on any computer (without python and the repo), however when I shared the .spec file for a colleague to build with (same repo and we are both using Visual Studios) he cannot build a working executable (numerous import errors). So far I have looked into sys.path and have modified the spec file to include each missing module by appending to datas, but there are a lot of modules missing so it is a cumbersome process and I do not need to do any of this on my computer. 
Any ideas as to what else to check?

**Comment by bwoodsend** [comment, marked as answer]
He's probably mixing up Python environments so that `pyinstaller` and his packages are in different environments. Get him to use `python -m` for pip and PyInstaller.
```
python -m pip install pyinstaller your dependencies here

# Check your code actually runs under this Python environment when invoked from cmd.
python your-code.py

# Do the build - note PyInstaller is case-sensitive here 
python -m PyInstaller your-spec-file.spec
```

**Reply by isisgonzalez183** [reply]
Thank you so much it worked like a charm! Note for anyone else reading this thread: He also uninstalled all packages from his machine and deleted the working virtual  environment. He then reinstalled all packages and the virtual environment with the python -m pip install flag.
